pkg/git/executor/executor.go

```diff
@@ -14,15 +15,17 @@ type Executor interface {
 
 // ShellExecutor an executor that shells out to run commands
 type ShellExecutor struct {
-	parentCtx context.Context
-	timeout   time.Duration
+	parentCtx         context.Context
+	timeout           time.Duration
+	privateSSHKeyPath string
```

Comment:
I think this executor, even though today it is only used for git, it could be reused for command (like for kubectl or anything else we need to shell out to in the future). So perhaps this field instead of being specific for the ssh key it could be something like environment variables. What do you think?

Reply:
I'd rather stay specific for now, but good to keep this in mind if it grows, yep 👍

Reply:
Let's fix it in another PR. There are also some git parameters grouping to be done

Reply:
Sure, I'll open a follow-up PR _soonishly_ then 👍

Reply:
Follow-up PR: https://github.com/weaveworks/eksctl/pull/1168

Reply:
Follow-up PR: https://github.com/weaveworks/eksctl/pull/1168